torch/_torch_docs.py

```diff
@@ -2536,6 +2536,32 @@ def merge_dicts(*dicts):
 
 """)
 
+add_docstr(torch.logical_not,
+           r"""
+logical_not(input, out=None) -> Tensor
+
+Computes the element-wise logical NOT of the given input tensor. The output tensor always has the same dtype as that of
```

Comment:
this is different than numpy's logical_not, which defaults to returning bool -- is there any reason why we don't do that?  Basically, I think this should behave the same way as our comparison operators.

Reply:
If we do this, can we still have inplace operator? What I'm thinking is that, in this case, a user should use `bool()` to do the conversion if a user needs bool tensors. This is also consistent with what `bitwise_not()` currently does.

Reply:
I think it should be behave like a comparison ops; in that case inplace keeps the dtype:
```
>>> x.gt_(1)
tensor([[0., 0., 1.],
        [0., 0., 0.]])
```

btw if this is difficult it seems fine to just support boolean for now (so we can get it into 1.2) and then add the other dtypes in a follow-on patch.

Reply:
OK, I'll change it to bool only for now so we can safely have it into v1.2.